File: articles/the-future-of-algorithms-computational-reasoning/go/automation_risk.go
package main

import "fmt"

func automationRisk(stakes, opacity, delegation, irreversibility float64) float64 {
	risk := stakes * opacity * delegation * irreversibility
	if risk < 0 { return 0 }
	if risk > 1 { return 1 }
	return risk
}

func main() {
	fmt.Printf("%.6f\n", automationRisk(0.95, 0.85, 0.90, 0.80))
}

File: articles/the-future-of-algorithms-computational-reasoning/haskell/AutomationRisk.hs
module Main where

automationRisk :: Double -> Double -> Double -> Double -> Double
automationRisk stakes opacity delegation irreversibility =
  max 0.0 (min 1.0 (stakes * opacity * delegation * irreversibility))

main :: IO ()
main = print (automationRisk 0.95 0.85 0.90 0.80)
